### PR TITLE
Thread-safe improvements in improvements in error handling

### PR DESCRIPTION
**Pull request by Usaga**
Errors now show in MainWindow's thread

### CODE REVIEW

**Comment by abma** [comment]
sorry, thats not thread safe. wx docs say:

http://docs.wxwidgets.org/trunk/classwx_evt_handler.html#a0737c6d2cbcd5ded4b1ecdd53ed0def3

"This function is similar to QueueEvent() but can't be used to post events from worker threads for the event objects with wxString fields (i.e. in practice most of them) because of an unsafe use of the same wxString object which happens because the wxString field in the original event object and its copy made internally by this function share the same string buffer internally. Use QueueEvent() to avoid this."

**Comment by Usaga** [comment]
Aghhh. Ok, will change it

**Comment by abma** [comment]
the official buildslave afaik uses wxwidgets 2.9.5.0. travis currently 2.8.12.1. so bumping wx minimal version to 2.9.5.0 is ok when travis is fixed.

this would allow to fix the same bug in the download search dialog.

**Comment by abma** [comment]
see #344

**Comment by Usaga** [comment]
Didn't saw difference in standalone test, but hope this will work fine

**Comment by abma** [comment]
FYI: https://github.com/springlobby/springlobby/commit/549444fdf2ab2c579280cd118966a81fdbc01bfa

File: src/gui/mainwindow.h [hunk 1]
```diff
@@ -196,6 +197,8 @@ class MainWindow : public wxFrame, public WindowAttributesPickle
         static const unsigned int OPT_PAGE_GROUPS   = 4;
 
 	const wxArrayString& GetTabNames();
+	
+	static const unsigned int mySHOW_ERROR_MESSAGE = 10000;
```
Comment (abma) [inline src/gui/mainwindow.h]: imo use wxNewId(); here

File: src/log.cpp [hunk 2]
```diff
@@ -70,7 +76,7 @@ class myLogger: public wxLog
 
 	std::string LogLevelToString(wxLogLevel level)
 	{
-		assert(level<8); // just in case
+	  assert(level<8); // just in case
```
Comment (abma) [inline src/log.cpp]: please use real tabs :)

Comment (Usaga) [inline src/log.cpp]: Ok))

File: src/gui/mainwindow.h (hunk 1 above)
Comment (Usaga) [inline src/gui/mainwindow.h]: http://docs.wxwidgets.org/trunk/group__group__funcmacro__misc.html#gaa2d93c1680369b4227f7de9de58e5fcb
Looks like this is not a good idea

Comment (abma) [inline src/gui/mainwindow.h]: oO
http://docs.wxwidgets.org/trunk/page_stdevtid.html

how to use it then?

Comment (Usaga) [inline src/gui/mainwindow.h]: May be myID = wxID_HIGHEST + $unique_number?

Comment (Usaga) [inline src/gui/mainwindow.h]: But there is no guarantee some one useses same $unique_number elsewhere in code in future

Comment (abma) [inline src/gui/mainwindow.h]: yeah, i understand it like that, too. wxID_HIGHEST + $unique_number should be fine then.

**Comment by abma** [comment]
thanks!

**Pull request merged**